src/databricks/labs/remorph/reconcile/compare.py

```diff
@@ -21,26 +22,53 @@ def raise_column_mismatch_exception(msg: str, source_missing: list[str], target_
 
 
 def reconcile_data(
-    source: DataFrame, target: DataFrame, key_columns: list[str], report_type: str
+    source: DataFrame,
+    target: DataFrame,
+    key_columns: list[str],
+    report_type: str,
+    spark: SparkSession | None = None,
+    path: str | None = None,
 ) -> DataReconcileOutput:
     source_alias = "src"
     target_alias = "tgt"
     if report_type not in {"data", "all"}:
         key_columns = [_HASH_COLUMN_NAME]
-    df = source.alias(source_alias).join(other=target.alias(target_alias), on=key_columns, how="full")
-
-    mismatch = (
-        _get_mismatch_data(df, source_alias, target_alias, source.columns) if report_type in {"all", "data"} else None
+    df = (
+        source.alias(source_alias)
+        .join(other=target.alias(target_alias), on=key_columns, how="full")
+        .selectExpr(
+            *[f'{source_alias}.{col_name} as {source_alias}_{col_name}' for col_name in source.columns],
+            *[f'{target_alias}.{col_name} as {target_alias}_{col_name}' for col_name in target.columns],
+        )
     )
+
+    if spark and path:
+        df = write_and_read_unmatched_df_with_volumes(df, spark, path)
+
+    mismatch = _get_mismatch_data(df, source_alias, target_alias) if report_type in {"all", "data"} else None
+
     missing_in_src = (
-        df.filter(col(f"{source_alias}.{_HASH_COLUMN_NAME}").isNull())
-        .select((key_columns if report_type == "all" else alias_column_str(target_alias, target.columns)))
-        .drop(_HASH_COLUMN_NAME)
+        df.filter(col(f"{source_alias}_{_HASH_COLUMN_NAME}").isNull())
+        .select(
+            *[
+                col(col_name).alias(col_name.replace(f'{target_alias}_', '').lower())
+                for col_name in df.columns
+                if col_name.startswith(f'{target_alias}_')
+            ]
+        )
+        .drop(f"{_HASH_COLUMN_NAME}")
```

Comment:
```suggestion
        .drop(_HASH_COLUMN_NAME)
```
F-string is redundant here

Reply:
Done